server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -572,8 +572,10 @@ public Index getResizeSourceIndex() {
 
     public static final String INDEX_ROLLUP_SOURCE_UUID_KEY = "index.rollup.source.uuid";
     public static final String INDEX_ROLLUP_SOURCE_NAME_KEY = "index.rollup.source.name";
-    public static final Setting<String> INDEX_ROLLUP_SOURCE_UUID = Setting.simpleString(INDEX_ROLLUP_SOURCE_UUID_KEY);
-    public static final Setting<String> INDEX_ROLLUP_SOURCE_NAME = Setting.simpleString(INDEX_ROLLUP_SOURCE_NAME_KEY);
+    public static final Setting<String> INDEX_ROLLUP_SOURCE_UUID = Setting.simpleString(INDEX_ROLLUP_SOURCE_UUID_KEY,
+        Property.IndexScope, Property.InternalIndex);
+    public static final Setting<String> INDEX_ROLLUP_SOURCE_NAME = Setting.simpleString(INDEX_ROLLUP_SOURCE_NAME_KEY,
+        Property.IndexScope, Property.InternalIndex);
```

Comment:
Are we sure we want `Property.InternalIndex` and not `Property.PrivateIndex`? I suppose the other setting that behaves like this is `index.xpack.ccr.following_index`. I'm not sure what the history is there, but I feel like we should discuss this. It feels leaky that we would allow this setting to be used during index creation.

Could we not find a reasonable way to use the ClusterService to create the index?